webapp/src/clj/lipas/backend/legacy/api.clj

```diff
@@ -0,0 +1,48 @@
+(ns lipas.backend.legacy.api
+  (:require
+   [lipas.data.prop-types :as prop-types]
+   [lipas.data.types :as types]
+   [lipas.utils :as utils]))
+
+(def keys-vec [:type-code :name :description :geometry-type :sub-category])
+
+(defn fill-properties [m]
+  (reduce (fn [acc k] (assoc-in acc [:props k] (prop-types/all k)))
+          m
+          (-> m :props keys)))
+
+
+(defn localize-properties [m locale]
+  (reduce (fn [acc k]
+            (-> acc
+                (update-in [:props k :description] locale)
+                (update-in [:props k :name] locale))) m (-> m :props keys)))
+
+(defn- localize [m lang]
+  (-> m
+      (update :name lang)
+      (update :description lang)))
+
+
+(defn- ->legacy-api-with-properties [m lang]
+  (-> m
+      (fill-properties)
+      (select-keys (conj keys-vec :props))
+      (update :props #(-> % (dissoc :schoolUse? :freeUse?)))
+      (localize-properties lang)
+      (localize lang)
+      utils/->camel-case-keywords))
+
+(defn- ->legacy-api [m lang]
+  (-> m
+      (select-keys keys-vec)
+      (localize lang)
+      utils/->camel-case-keywords))
```

Comment:
It would be nice if it was this simple. ;) However, the differences between lipas and legacy-api are not 1:1 kebab-case -> camelCase... there are a few edge-cases and exceptions that are covered in this mapping: https://github.com/lipas-liikuntapaikat/lipas/blob/master/webapp/src/clj/lipas/integration/old_lipas/sports_site.clj#L13

You can use [rename-keys](https://clojuredocs.org/clojure.set/rename-keys) to change "new keys" to "old keys" with the hardcoded map.